roles/reproducer/tasks/devscripts_layout.yml

```diff
@@ -1,6 +1,224 @@
+      {{
+        _pub_net_data.network_v6 | ansible.utils.nthhost(3)
+      }}
+      {% endif -%}
+  ansible.builtin.copy:
+    dest: "/etc/cifmw-dnsmasq.d/ocp_dns.conf"
+    mode: "0644"
+    content: |
+      local=/{{ _cluster }}.openstack.lab/
+      domain={{ _cluster }}.openstack.lab
+      address=/api.{{ _cluster }}.openstack.lab/{{ _api_addr }}
+      address=/api-int.{{ _cluster }}.openstack.lab/{{ _api_addr }}
+      address=/.apps.{{ _cluster }}.openstack.lab/{{ _apps_addr }}
+      localise-queries
+    validate: "/usr/sbin/dnsmasq -C %s --test"
+
+- name: Configure NetworkManager
+  become: true
+  notify: Restart NetworkManager
+  ansible.builtin.copy:
+    dest: >-
+      /etc/NetworkManager/conf.d/{{ cifmw_reproducer_networkmanager_config }}
+    mode: "0644"
+    content: |
+      [main]
+      rc-manager=none
+      dns=none
+    validate: "NetworkManager --print-config %s"
+
+- name: Ensure services are reloaded
+  ansible.builtin.meta: flush_handlers
+
+- name: Ensure we use dnsmasq as resolver
+  become: true
+  vars:
+    _cluster: "{{ cifmw_devscripts_config['cluster_name'] }}"
+  ansible.builtin.copy:
+    dest: /etc/resolv.conf
+    content: |
+      search {{ _cluster }}.openstack.lab
+      nameserver {{ _libvirt_manager_networking.networks.ctlplane.gw_v4 }}
+
+- name: Bootstrap devscript using baremetal provisioning
+  when:
+    - not cifmw_devscripts_ocp_exists | bool
+  vars:
+    _bm_ironic_file: >-
+      {{
+        [cifmw_reproducer_basedir,
+         'artifacts', 'baremetal-ironic.json'] | path_join
+      }}
+  block:
+    - name: Create blank images for OCP cluster
+      ansible.builtin.import_role:
+        name: "libvirt_manager"
+        tasks_from: "ocp_layout.yml"
+
+    - name: Refresh VBMC data
+      ansible.builtin.import_role:
+        name: "virtualbmc"
+        tasks_from: "list_hosts.yml"
+
+    - name: Generate baremetal content fact
+      vars:
+        _hypervisor_ip4: >-
+          {{ hostvars[inventory_hostname].ansible_default_ipv4.address }}
+        _vm_data: "{{ _cifmw_libvirt_manager_layout.vms.ocp }}"
+        _mac: >-
+          {{
+            cifmw_libvirt_manager_mac_map[item] |
+            selectattr('network', 'equalto', cifmw_reproducer_provision_net) |
+            map(attribute='mac') | first
+          }}
+        _matcher: >-
+          {{ cifmw_devscripts_config.cluster_name }}_master_{{ vm_id }}
+        _ipmi_port: >-
+          {{
+            cifmw_virtualbmc_known_hosts |
+            selectattr('Domain name', 'match', _matcher) |
+            map(attribute='Port') | first
+          }}
+        _node:
+          name: "openshift-master-{{ vm_id  }}"
+          driver: "ipmi"
+          resource_class: "baremetal"
+          driver_info:
+            username: "admin"
+            password: "password"
+            address: "ipmi://{{ _hypervisor_ip4 }}:{{ _ipmi_port }}"
+            deploy_kernel: "http://172.22.0.2/images/ironic-python-agent.kernel"
```

Comment:
This needs to be configurable.

Reply:
I have a proposal in another draft PR #1817 we'll use to test NFV - apparently there are some needs in there since the range isn't the same.